### DISCUSSION TITLE
Configuring the Tendermint version

### DISCUSSION THREAD

**Discussion by mzabaluev** [informalsystems/hermes#3068] [Ideas]
I'm writing an adaptation of Hermes to multi-version Tendermint API in https://github.com/informalsystems/hermes/pull/3017

Is Tendermint version something we'd like to have configurable via config.toml (in `ChainConfig`), or would it be always automatically discovered with (as of 0.37) querying the `/status` RPC endpoint and looking at the `.node_info.version` field of the response?

NB: the `/status` response technically should already conform to a certain RPC spec version to be reliably parsed, but it's a subject for another issue.

**Comment by adizere** [comment]
Thanks for kicking off this conversation Mikhail.

I think making the Tendermint version configurable is a potential source of headaches for operators. I would not expect the typical IBC relayer operator to necessarily know the Tendermint version of all networks they're relaying on.

Also, Tendermint is a network-specific building block. Different networks will have different underlying consensus algorithms.

> would it be always automatically discovered with (as of 0.37) querying the /status RPC endpoint and looking at the .node_info.version field of the response?

This sounds preferable. Do you see any disadvantage to this approach?

**Comment by ancazamfir** [comment]
> Is Tendermint version something we'd like to have configurable via config.toml (in ChainConfig), ...

No.
> or would it be always automatically discovered with (as of 0.37) querying the /status RPC endpoint and looking at the .node_info.version field of the response?

Yes
